CPS-????/README.md

```diff
@@ -0,0 +1,81 @@
+
+### The Current Alternative
+The alternative involves taking advantage of the fact that staking scripts do not suffer from the same limitation as spending scripts.
+
+> Given a [spending] script **s**, you can turn it into a [spending] script **s'** that checks that 0 ADA is withdrawn from a [staking] script **s''** with the same logic as **s**. This is a transformation that is valid for any [spending] script.    -- @L-as
+
+In a nut shell, you use a staking script **AS** a spending script. The transaction level logic would be in the staking script. Meanwhile, the spending script just checks that both the required staking script is executed AND 0 ADA is withdrawn from the rewards address within the transaction. The withdrawal of the 0 ADA is necessary to force the staking script to be executed.
+
+The only requirement to use this technique is that the script address must have been created using the `s'` spending script and the `s''` staking script. The staking address DOES NOT need to actually have rewards or even be delegated/registered for this technique to work.
+
+With this method, the main computation is only executed once no matter how many UTxOs are spent from the script address.
+
+#### The Drawbacks
+
+1. This technique now means that everything requires two plutus scripts instead of just one. While the spending script can be made extremely small, this doesn't change the fact that two are now needed.
+2. This technique DOES NOT stop the redundant executions; the spending script will still redundantly check if the required staking script was executed. This technique just minimizes the cost of the redundant executions by moving the heavy computation into a script that is only executed once per transaction.
+3. This is not the intended use for staking scripts.
+
+### Simple Scripts
+I tried to test Cardano simple scripts to determine if this problem also impacts them. However, given that simple scripts are so small, it was hard to tell from just looking at the transaction fee. Since the interface for using Cardano simple scripts is the same as for using plutus scripts, I have to assume that this problem also impacts simple scripts. 
+
+If simple scripts are indeed impacted by this problem, the above technique would not work for them. The technique only works for plutus scripts.
+
+## Use Cases
+Any time a spending script's validation depends on the entire transaction and not on any particular UTxO.
+
+1. P2P Atomic Swaps - within one transaction, the value entering the swap address must be some proportion of the value leaving the swap address
```

Comment:
Would it be fair to ask for a more detailed explanation of this use case?  This may be naive (and I'm not technically capable of joining the technical debate of the proposal itself) but if a treatment of this issue is helpful in implementing atomic swaps on Cardano, then I believe this use case should be covered in detail in the document... to progress this CPS and any CIP that facilitates that treatment.

I'm trying to get more reviewers based on interest in the Atomic Swap idea as you've described it on GitHub.  If in agreement with what I've said above, I also think this repository should be linked from the CPS itself.  Here is what I posted about [fallen-icarus/cardano-swaps](https://github.com/fallen-icarus/cardano-swaps) (please correct me if I've described the issue inaccurately or naïvely) in the Cardano MBO on Matrix

> I've been watching this concept develop from https://github.com/cardano-foundation/CIPs/pull/418 and I don't think third party reviewers have been giving it enough credit, or even enough attention.  I don't know every technical underpinning of every current & proposed DEX on Cardano but in the meantime this idea seems unique to me.  I do wish I could do more to promote this idea since my own technical shortcomings make it impossible for me to convincingly push ahead any proposed standards that would be required to make this happen.  I can only hope key people at IOG, CF, and Cardano dev agencies can take a serious look at how significant it would be for the industry & Cardano's own marketplace to support a true non-custodial DEX. 😯

https://matrix.to/#/!QvtUklhUfmEuCBIexz:matrix.org/$lU0OwYOAQXpNYVFJl1I33YPhhQ_wH14Jzu5etQP9Drk?via=matrix.adamantium.online&via=matrix.org&via=forum.balanceanalytics.io